test/parallel/test-dns-promise-channel-cancel.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+const common = require('../common');
+const dnstools = require('../common/dns');
+const { promises: dnsPromises } = require('dns');
+const assert = require('assert');
+const dgram = require('dgram');
+
+const server = dgram.createSocket('udp4');
+const resolver = new dnsPromises.Resolver();
+
+server.bind(0, common.mustCall(async () => {
+  resolver.setServers([`127.0.0.1:${server.address().port}`]);
+
+  await assert.rejects(
+    resolver.resolve4('example.org'),
+    {
+      code: 'ECANCELLED',
+      syscall: 'queryA',
+      hostname: 'example.org'
+    }
+  );
```

Comment:
Would be good to have multiple Promises in flight to ensure that all are canceled as expected.

Reply:
The same for callbacks as well.